storey/table.py

```diff
@@ -235,11 +251,122 @@ def __getitem__(self, key):
+                    await self._persist(_PersistJob(key, None, None))
+            await asyncio.sleep(self._flush_interval_milli / 1000)
+
+    async def _persist_worker(self):
+        task = None
+        received_job_count = 0
+        self_sent_jobs = {}
+        try:
+            while True:
+                jobs = self_sent_jobs.pop(received_job_count, None)
+                if jobs:
+                    task = jobs[0]
+                    if len(jobs) > 1:
+                        self_sent_jobs[received_job_count] = jobs[1:]
+                else:
+                    task = await self._q.get()
+                    received_job_count += 1
+                    if task is _termination_obj:
+                        if received_job_count in self_sent_jobs:
+                            await self._q.put(_termination_obj)
+                            continue
+                        for key, pending_event in self._pending_by_key.items():
+                            if pending_event.pending and not pending_event.in_flight:
+                                for job in pending_event.pending:
+                                    resp = await self._persist_key(key, job.data)
+                                    if job.callback:
+                                        await job.callback(job.extra_data, resp)
+                        break
+
+                job = task[0]
+                completed = await task[1]
+
+                for done_job in self._pending_by_key[job.key].in_flight:
+                    if done_job.callback:
+                        await done_job.callback(done_job.extra_data, completed)
+                self._pending_by_key[job.key].in_flight = []
+
+                # If we got more pending events for the same key process them
+                if self._pending_by_key[job.key].pending:
+                    self._pending_by_key[job.key].in_flight = self._pending_by_key[job.key].pending
+                    self._pending_by_key[job.key].pending = []
+
+                    future_task = self._safe_process_events(self._pending_by_key[job.key].in_flight)
+                    tail_position = received_job_count + self._q.qsize()
+                    jobs_at_tail = self_sent_jobs.get(tail_position, [])
+                    jobs_at_tail.append((job, asyncio.get_running_loop().create_task(future_task)))
+                    self_sent_jobs[tail_position] = jobs_at_tail
+                else:
+                    del self._pending_by_key[job.key]
+        except BaseException as ex:
+            if task and task is not _termination_obj and task[0].extra_data and task[0].extra_data._awaitable_result:
+                task[0].extra_data._awaitable_result._set_error(ex)
+            if not self._q.empty():
+                await self._q.get()
+            raise ex
+
+    async def _terminate(self):
+        if self._q:
+            self._terminated = True
+            last_flush = time.monotonic() - (self._flush_interval_milli * 1000)
```

Comment:
Hm, millis * 1000? Should have been division?

Reply:
Also, since we divide it everywhere, maybe we can ask for seconds to begin with? (in the constructor parameter)
Otherwise, at least we can convert it to seconds one time when we get it.